### PR TITLE
 Slash oracle bound while resolving disputed market, if oracle does not report or reports wrong outcome

### LINKED ISSUE #845: Oracle bond is not slashed when market resolves after dispute and oracle did not report
This is a bug in `on_resolution` which can easily be fixed. A test verifying the correct behavior should also be added. If you like, you could also use the opportunity to split the large `match` statement in the middle of this functions into helper functions.

### PR DESCRIPTION
**Pull request by vivekvpandya**
Fixes #845

### CODE REVIEW

**Review by Chralt98 — APPROVED** [review]
Nice test improvements for disputed markets 😉.

File: zrml/prediction-markets/src/tests.rs [hunk 1]
```diff
@@ -3203,6 +3159,50 @@ fn deploy_swap_pool_for_market_returns_error_if_weights_is_too_long() {
...
+            OutcomeReport::Categorical(0)
+        ));
+        run_to_block(grace_period + market.deadlines.dispute_duration + 1);
+        assert_eq!(Balances::reserved_balance(&ALICE), SENTINEL_AMOUNT);
+        assert_eq!(
+            Balances::free_balance(&ALICE),
+            alice_balance_before + ValidityBond::get() + OracleBond::get()
+        );
+    });
+}
```
[the remark this answers is not in the document]
Comment (vivekvpandya) [reply zrml/prediction-markets/src/tests.rs:3204]: just to keep similar tests together.

File: zrml/prediction-markets/src/tests.rs [hunk 2]
```diff
@@ -3328,6 +3328,319 @@ fn on_resolution_correctly_reserves_and_unreserves_bonds_for_approved_advised_ma
     });
 }
 
+#[test]
+fn on_resolution_correctly_reserves_and_unreserves_bonds_for_permissionless_market_with_correct_disputed_outcome_with_oracle_report()
```
[the remark this answers is not in the document]
Comment (vivekvpandya) [reply zrml/prediction-markets/src/tests.rs:3332]: :pray:

**Review by Chralt98 — APPROVED** [review]

**Review by maltekliemann — APPROVED** [review]
You'll find that some tests have been duplicated (see `on_resolution_defaults_to_oracle_report_in_case_of_unresolved_dispute`, for example), but this is something that we don't have time to consolidate right now, and your tests seem quite clear.

**Pull request merged**